### ISSUE TITLE
Nested functions: yield and raise wrongly attributed to parent function

### ISSUE THREAD

**Issue by AdamGleave** [terrencepreilly/darglint#132]
`darglint` expects "Yield" and "Raises" documentation to be in the parent of a nested function, rather than in the child as I would intuitively expect. In other respects it handles nested functions correctly.

Minimal example:
```python
def f(risky: bool):
    """I'm a thunk.

    Args:
        risky: Do you feel lucky, coder?

    Returns:
        A function! Woah.
    """

    def g(raises: bool):
        """I yield the truth, or end all.

        Raises:
            RuntimeError: if raises True.

        Yield:
            The truth."""
        if raises:
            raise RuntimeError("THE END.")
        yield 42

    def h(x: bool):
        """I always return the truth."""
        del x
        return 42

    if risky:
        return g
    else:
        return h
```

Output:
```
$ darglint test.py
test.py:f:1: DAR301: - yield
test.py:f:1: DAR401: -r RuntimeError
test.py:g:11: DAR101: - raises
test.py:g:11: DAR301: - yield
$ darglint --version
1.5.5
```

Nice linter by the way, it's already caught some errors in my documentation :+1:

**Comment by terrencepreilly** [comment]
This should be resolved in 2e1821a9afe168a341bad5f9cac5dcfa803c5912, which should be included in the 1.5.6 release.  Sorry it took a while: I spent some time and refactored how functions and methods are analyzed at the same time.  Thanks for submitting the issue.

**Comment by AdamGleave** [comment]
Thanks for the fix, I can confirm this resolves the issue, both in the example provided and in the project I'm using darglint in.

**Comment by terrencepreilly** [comment]
This should now be available in v1.5.6.
